helix-term/src/ui/mod.rs

```diff
@@ -333,7 +338,29 @@ pub mod completers {
         static KEYS: Lazy<Vec<String>> = Lazy::new(|| {
             let mut keys = Vec::new();
             let json = serde_json::json!(Config::default());
-            get_keys(&json, &mut keys, None);
+            get_keys(&json, &mut keys, None, false);
+            keys
+        });
+
+        let matcher = Matcher::default();
+
+        let mut matches: Vec<_> = KEYS
+            .iter()
+            .filter_map(|name| matcher.fuzzy_match(name, input).map(|score| (name, score)))
+            .collect();
+
+        matches.sort_unstable_by_key(|(_file, score)| Reverse(*score));
+        matches
+            .into_iter()
+            .map(|(name, _)| ((0..), name.into()))
+            .collect()
+    }
+
+    pub fn boolean_setting(_editor: &Editor, input: &str) -> Vec<Completion> {
```

Comment:
i would have preferred adding a helper function used by both `setting` and `boolean_setting`, but the `static KEYS` can't depend on a function parameter.

Reply:
you just have two `static` in the `setting` function:

``` rust
static KEYS
```

and 

``` rust
static KEYS_BOOL
```

and dispatch which of those two to use with a simple `if`. Not a perfect solution either but would reduce code duplication slightly. I don't imagine that we would add more specialized settings keys in the future as more advanced dynamic configuration is better achieved with a future plugin system. Until then hardocind the bool keys seems like a good enough solution.

Reply:
done in a separate commit, i had to duplicate the `.iter()` call as well